src/go/docker-credential-none/cmd/root.go

```diff
@@ -51,13 +52,22 @@ using the same format that docker uses when no credsStore field is specified in
 This helper is intended for testing purposes, but will be used on Linux systems
 unless 'pass' and/or 'secretservice' is available.`,
 		configFileName),
+	DisableSuggestions: true,
+	SilenceUsage:       true,
 }
 
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
 	err := rootCmd.Execute()
 	if err != nil {
+		ptn := regexp.MustCompile(`unknown command "(.*?)" for "(?i:docker-credential-none(?:\.exe)?)"`)
```

Comment:
Running it locally, it doesn't look like `.exe` ever shows up:
```
> .\docker-credential-none.exe aaa
Error: unknown command "aaa" for "docker-credential-none"
Run 'docker-credential-none --help' for usage.
Unknown credential action `aaa`
```

Also, note that we're still emitting the normal error _in addition to_ your custom one, rather than just the custom error; contrast with `wincred`:
```
> .\docker-credential-wincred.exe aaa
Unknown credential action `aaa`
```

(While I prefer not doing string matching, it doesn't look like that's reasonable given the cobra code:
https://github.com/spf13/cobra/blob/45e521ea93e2270ef7333487f0a87e3887de2da9/args.go#L22
We might need to just not use cobra here, instead…)

Reply:
I agree ... there is exactly one argument to process here, and cobra is getting in the way.

It might be a while before I get back to this, given the offline performance is a priority.

Reply:
I added our own `os.Args` checker before handing it over to Cobra. The results look more conformant with the other helpers.